arm_robots/src/arm_robots/panda.py

```diff
@@ -15,11 +21,111 @@ def __init__(self, robot_namespace: str = 'combined_panda', force_trigger: float
                                     arms_controller_name='effort_joint_trajectory_controller',
                                     force_trigger=force_trigger,
                                     **kwargs)
+        self.nebula_arm = "nebula_arm"
+        self.nebula_id = "panda_1"
+        self.rocket_arm = "rocket_arm"
+        self.rocket_id = "panda_2"
+        self.nebula_wrist = "panda_1_link8"
+        self.rocket_wrist = "panda_2_link8"
+        self.nebula_gripper = PandaGripper(self.robot_namespace, self.nebula_id)
+        self.rocket_gripper = PandaGripper(self.robot_namespace, self.rocket_id)
+        rospy.Subscriber("/"+self.robot_namespace+"/"+self.nebula_id+"_state_controller/franka_states", FrankaState, self.nebula_error_cb)
+        rospy.Subscriber("/"+self.robot_namespace+"/"+self.rocket_id+"_state_controller/franka_states", FrankaState, self.rocket_error_cb)
+
+    def open_nebula_gripper(self):
+        self.nebula_gripper.move(self.nebula_gripper.MAX_WIDTH)
+
+    def close_nebula_gripper(self):
+        self.nebula_gripper.move(self.nebula_gripper.MIN_WIDTH)
+
+    def open_rocket_gripper(self):
+        self.rocket_gripper.move(self.rocket_gripper.MAX_WIDTH)
+
+    def close_rocket_gripper(self):
+        self.rocket_gripper.move(self.rocket_gripper.MIN_WIDTH)
+
+    def nebula_error_cb(self, data):
+        if data.current_errors.communication_constraints_violation or data.last_motion_errors.communication_constraints_violation:
+            self.clear_errors(wait_for_result=True)
 
-    def send_joint_command(self, joint_names: List[str], trajectory_point: JointTrajectoryPoint) -> Tuple[bool, str]:
-        # TODO: Fill in to send set point to controller.
-        pass
+    def rocket_error_cb(self, data):
+        if data.current_errors.communication_constraints_violation or data.last_motion_errors.communication_constraints_violation:
+            self.clear_errors(wait_for_result=True)
 
-    # TODO: Add gripper helpers.
+    def clear_errors(self, wait_for_result=False):
+        self.error_client = actionlib.SimpleActionClient("/"+self.robot_namespace+"/error_recovery", ErrorRecoveryAction)
```

Comment:
consider removing the leading `/`

Reply:
also consider moving this to the constructor, as well as the wait_for_server call